pallets/did/src/lib.rs

```diff
@@ -592,6 +618,41 @@ pub mod pallet {
 			//TODO: Return the real weight used
 			Ok(().into())
 		}
+
+		/// Deletes all the information associated with a DID on chain, after
+		/// verifying the signature associated with the operation. The
+		/// parameters are:
+		/// * origin: the Substrate account submitting the transaction (which
+		///   can be different from the DID subject)
+		/// * did_deletion_operation: a DidDeletionOperation which includes the
+		///   DID to deactivate
+		/// * signature: a signature over the operation that must be signed with
+		///   the authentication key associated with the new DID.
+		#[pallet::weight(<T as Config>::WeightInfo::submit_did_delete_operation())]
+		pub fn submit_did_delete_operation(
+			origin: OriginFor<T>,
+			did_deletion_operation: DidDeletionOperation<T::DidIdentifier>,
+			signature: DidSignature,
+		) -> DispatchResultWithPostInfo {
+			// origin of the transaction needs to be a signed sender account
+			let sender = ensure_signed(origin)?;
+
+			let did_identifier = did_deletion_operation.get_did();
+
+			// If specified DID does not exist, generate a DidNotPresent error.
+			let did_details = <Did<T>>::get(&did_identifier).ok_or(<Error<T>>::DidNotPresent)?;
+
+			// Verify the signature of the delete operation.
+			Self::verify_operation_signature_for_entry(&did_deletion_operation, &signature, &did_details)
+				.map_err(<Error<T>>::from)?;
+
+			log::debug!("Deleting DID {:?}", did_identifier);
+			<Did<T>>::take(&did_identifier);
+
+			Self::deposit_event(Event::DidDeleted(sender, did_identifier.to_owned()));
+			//TODO: Return the real weight used
```

Comment:
No need to adjust the weight after execution since the number of read/writes is constant.

Reply:
Does this apply also to the other extrinsics in the DID pallet?

Reply:
yes. You only need to adjust the weight if you have DB read/writes in a if/foreach-branch.